spec/jobs/detect_posts_translation_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
```

Comment:
nitpick but I would probably do this:

```
let(:redis_key) { DiscourseTranslator::LANG_DETECT_NEEDED }
```

This will make the spec less heavy/redundant to parse